nativelib/src/main/resources/gc/immix/Stats.c

```diff
@@ -0,0 +1,45 @@
+#include "Stats.h"
+#include <stdio.h>
+
+void Stats_writeStatsToFile(Stats *stats);
+
+void Stats_Init(Stats *stats, const char *statsFile) {
+    stats->outFile = fopen(statsFile, "a");
+    fprintf(stats->outFile, "timestamp_ns,collection,mark_time_ns,sweep_time_ns\n");
+    stats->collections = 0;
+}
+
+void Stats_RecordCollection(Stats *stats, uint64_t start_ns, uint64_t sweep_start_ns, uint64_t end_ns) {
+    uint64_t index = stats->collections % GC_STATS_MEASUREMENTS;
+    stats->timestamp_ns[index] = start_ns;
+    stats->mark_time_ns[index] = sweep_start_ns - start_ns;
+    stats->sweep_time_ns[index] = end_ns - sweep_start_ns;
```

Comment:
I don't think we need timestamp_ns at all, mark and sweep times and collection # are more than sufficient on their own.

Reply:
In fact collection # is redundant as well, it can be trivially recomputed as line number in the output file.

Reply:
I would rather keep these two.

Yes, when calculating aggregates for benchmarks, `collection #` and `timestamp` are redundant. However, I wanted to see/show the difference between a few large collections and many smaller ones by using the timestamps.

I also considered how useful it may be for diagnostics. Timestamps are important as they allow correlating other events. `collection #` allows getting a piece of the file and still making sense what's happening. The output file is append mode, so getting the `collection #` from the line number can be not so straight forward.

Reply:
Timestamp is a poor substitute for this as youd'd want to correlate measurements with the iterations of the benchmark for it to be meaningful. Mutator time can be different as well, making the relative comparisons based on timestamps to be completely off.

Reply:
It's also safe to assume that append case effectively never happens. If someone cares about more separation of data they can pass a different file on every invocation.